File: card-api/src/test/java/com/jean202/cardmizer/api/demo/RecommendationDemoScenariosLoaderTest.java
package com.jean202.cardmizer.api.demo;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

class RecommendationDemoScenariosLoaderTest {
    @Test
    void loadsDefaultRecommendationDemoScenarios() {
        RecommendationDemoScenarios scenarios = new RecommendationDemoScenariosLoader().loadDefault();

        assertEquals(4, scenarios.scenarios().size());
        assertTrue(scenarios.scenarios().stream().anyMatch(scenario -> scenario.id().equals("kpass-transit-threshold")));
        assertTrue(scenarios.scenarios().stream().allMatch(scenario -> !scenario.seedRecords().isEmpty()));
    }
}

File: card-api/src/test/kotlin/com/jean202/cardmizer/api/demo/RecommendationDemoScenariosLoaderTest.kt
package com.jean202.cardmizer.api.demo

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class RecommendationDemoScenariosLoaderTest {
    @Test
    fun loadsDefaultRecommendationDemoScenarios() {
        val scenarios = RecommendationDemoScenariosLoader().loadDefault()

        assertEquals(4, scenarios.scenarios.size)
        assertTrue(scenarios.scenarios.any { it.id == "kpass-transit-threshold" })
        assertTrue(scenarios.scenarios.all { it.seedRecords.isNotEmpty() })
    }
}
